gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -1220,24 +1187,26 @@ gst_tensor_filter_transform_caps (GstBaseTransform * trans,
       config.info = self->prop.output_meta;
       result = gst_tensor_filter_caps_from_config (self, &config);
     } else {
-      GstTensorsInfo in_info;
-
       /* check in-tensor info to call setInputDimension */
-      in_info = config.info;
-      if (gst_tensors_info_validate (&in_info)) {
+      if (gst_tensors_info_validate (&config.info)) {
+        GstTensorsInfo out_info;
         int res = -1;
 
         /* call setInputDimension with given input tensor */
-        gst_tensor_filter_call (self, res, setInputDimension, &in_info,
-            &config.info);
+        gst_tensors_info_init (&out_info);
+        gst_tensor_filter_call (self, res, setInputDimension, &config.info,
+            &out_info);
 
         if (res == 0) {
+          config.info = out_info;
```

Comment:
what is the purpose of this memcpy of the two structs?
`out_info` is initialized with line 1196 and its dimension set with 1197 using `config.info`. As far as I understand, out_info has no extra information.

Further, memory for `out_info->info[]->name` is freed in 1209. If we assign this name to `config.info`, `config.info` name would also be freed.

Reply:
1. vmethod 'setInputDimension' can be used if subplugin should determine output tensor dimension using input-tensor info. (e.g., input with single tensor 1:200:1:1, output with multi-tensors 1:200:1:1, 1:1:1:1)
So when called setInputDimension in line 1197, filter-subplugin will fill the out_info and out_info may have the name string.

We don't use name for the caps negotiation. So config.info name (result from gst_tensors_config_from_structure) is null.

Reply:
Thanks for the explanation.